pkg/scheduler/framework/v1alpha1/waiting_pods_map.go

```diff
@@ -85,6 +85,10 @@ func newWaitingPod(pod *v1.Pod, pluginsMaxWaitTime map[string]time.Duration) *wa
 	}
 
 	wp.pendingPlugins = make(map[string]*time.Timer, len(pluginsMaxWaitTime))
+	// The time.AfterFunc calls wp.Reject which iterates through pendingPlugins map. Acquire the
+	// lock here so that time.AfterFunc can only execute after newWaitingPod finishes.
+	wp.mu.Lock()
+	defer wp.mu.Unlock()
```

Comment:
I think it is ok for time.AfterFunc to execute before newWaitingPod finishes, because when `newWaitingPod` is called, it means that permit plugins has been running and we could start timer for them. 

Is there any other concern?

Reply:
Yes this causes race. In time.AfterFunc,  `wp.Reject` is called, which needs to iterate through the map. I ran into that in tests.

Reply:
SG